### DISCUSSION TITLE
JMusicBot doesn't reconnect after internet outage

### DISCUSSION THREAD

**Discussion by Randy-Peeters** [jagrosh/MusicBot#938] [Troubleshooting]
### Is your feature request related to a problem? Please describe.

Im not sure if i'm right but i think that if there is no internet connection there will be an issue. So that seems logical but the real issue is that it does't retry to connect. I got these error messages:

Sep 30 08:40:24 ubuntu run.sh[1716]:         at net.dv8tion.jda.api.requests.RestFuture.<init>(RestFuture.java:36)
Sep 30 08:40:24 ubuntu run.sh[1716]:         at net.dv8tion.jda.internal.requests.RestActionImpl.submit(RestActionImpl.java:209)
Sep 30 08:40:24 ubuntu run.sh[1716]:         at net.dv8tion.jda.internal.requests.RestActionImpl.complete(RestActionImpl.java:219)
Sep 30 08:40:24 ubuntu run.sh[1716]:         at net.dv8tion.jda.api.requests.RestAction.complete(RestAction.java:633)
Sep 30 08:40:24 ubuntu run.sh[1716]:         at net.dv8tion.jda.api.utils.SessionControllerAdapter.getGateway(SessionControllerAdapter.java:86)
Sep 30 08:40:24 ubuntu run.sh[1716]:         at net.dv8tion.jda.internal.JDAImpl.getGateway(JDAImpl.java:318)
Sep 30 08:40:24 ubuntu run.sh[1716]:         at net.dv8tion.jda.internal.JDAImpl.login(JDAImpl.java:276)
Sep 30 08:40:24 ubuntu run.sh[1716]:         at net.dv8tion.jda.internal.JDAImpl.login(JDAImpl.java:268)
Sep 30 08:40:24 ubuntu run.sh[1716]:         at net.dv8tion.jda.api.JDABuilder.build(JDABuilder.java:1870)
Sep 30 08:40:24 ubuntu run.sh[1716]:         at com.jagrosh.jmusicbot.JMusicBot.main(JMusicBot.java:181)

### What is your ideal solution to the problem?

make sure that the program tries to reconnect after an error that there is no internet.

### How would this feature be used?

It is an actual problem to be fixed so everyone can use this feature.

### Checklist

- [X] I have checked [the documentation](https://jmusicbot.com/) to make sure this feature doesn't already exist
- [X] I have searched for [similar feature requests](https://github.com/jagrosh/MusicBot/labels/enhancement)
- [X] I have confirmed that my requests is **not** on the [list of things that will not be added](https://jmusicbot.com/things-that-wont-be-added)
- [x] I am running the latest version of the bot: ![Release](https://img.shields.io/github/release/jagrosh/MusicBot.svg)

**Comment by BrunoBerger** [comment]
I have the same problem when the bot tries to connect after boot but my server has no connection yet.
It would be helpfull if the bot would either try again or atleast exit properly, so it can autimatically restart.
